common/XenomaiLock.h

```diff
@@ -0,0 +1,42 @@
+#include <mutex> //unique_lock
+// Xenomai enforces the requirement to hold the lock so that optimizing
+// the wake up process is possible - under the assumption that the caller
+// owns the lock. There is no hack around this.
+// See: https://www.xenomai.org/pipermail/xenomai/2017-October/037759.html
+#define SC_CONDITION_VARIABLE_ANY_SHOULD_LOCK_BEFORE_NOTIFY
```

Comment:
The name of the macro describes the behaviour implemented in `common/SC_SyncCondition.h` when the macro is defined. It is a requirement for Xenomai that the calling thread holds the lock before notifying the waiting thread. There may be other APIs/systems where this is needed (although I don't know about any), so I tried to keep the name and logic Xenomai-independent. Maybe I was trying to be too clever?

Reply:
Personally I understand your logic, and agree with it, thanks for explaining :) Maybe I was trying to be too clever here. Just pointing this out, in case anyone else has objections, and if not, for me it can stay as it is.